lib/zaikio/client/helpers/authorization_middleware.rb

```diff
@@ -0,0 +1,32 @@
+require "faraday"
+require "jwt"
+require "concurrent"
+
+module Zaikio
+  module Client
+    module Helpers
+      class AuthorizationMiddleware < Faraday::Middleware
+        def self.token
+          @token ||= Concurrent::ThreadLocalVar.new { nil }
+          @token.value
+        end
+
+        def self.token=(value)
+          @token ||= Concurrent::ThreadLocalVar.new { nil }
+          @token.value = value
+        end
+
+        def self.reset_token
+          self.token = nil
+        end
+
+        def call(request_env)
+          request_env[:request_headers]["Authorization"] = "Bearer #{self.class.token}" if self.class.token
+
+          @app.call(request_env).on_complete do |response_env|
+          end
```

Comment:
Does this need an empty response handler, or does it work without?

```suggestion
          @app.call(request_env)
```

Reply:
just copied, I did not put too much thought in it as we can see